dhis-2/dhis-services/dhis-service-tracker/src/test/java/org/hisp/dhis/tracker/validation/hooks/PreCheckExistenceValidationHookTest.java

```diff
@@ -505,6 +508,22 @@ void verifyRelationshipValidationFailsWhenIsDeleteAndRelationshipIsNotPresent()
         hasTrackerError( reporter, E4016, RELATIONSHIP, rel.getUid() );
     }
 
+    @Test
+    void verifyRelationshipValidationFailsWhenIsSoftDeleted()
+    {
+        // given
```

Comment:
I don't think we need these comments. Your test is very readable 😄 with the arrange, act, assert groups.

Reply:
Removed